src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -217,6 +240,9 @@ protected override void SetParentCore([DynamicallyAccessedMembers(DynamicallyAcc
         public override string Name => _name;
         public override Type? DeclaringType => _declaringType;
         public override Type? ReflectedType => _declaringType;
+        public override bool IsGenericTypeDefinition => IsGenericType;
+        public override bool IsGenericType => _typeParameters != null;
+        public override Type[] GenericTypeParameters => _typeParameters ?? EmptyTypes;
```

Comment:
Also, this should override `GetGenericArguments`. Overriding `GenericTypeParameters` is nice-to-have.